--- helper_func_with_printf.c	2024-05-07 20:39:03.093301008 +0200
+++ helper_func.c	2024-05-07 20:35:57.951426335 +0200
@@ -5,6 +5,7 @@
 #include <stdio.h>
 #include <stdbool.h>
 #include <time.h>
+#include <unistd.h>
 #include "constants.h"
 #include "err.h"
 
@@ -96,11 +97,10 @@
 }
 
 // - Function prints buff_len bytes from buff to stdout, function flushes stdout
-// to make sure printf() prints wanted bytes
+// to make sure write() prints wanted bytes
 void print_data_to_stdout(char *buff, uint32_t buff_len)
 {
-    printf("%.*s", (int)buff_len, buff);
-    fflush(stdout);
+    write(STDOUT_FILENO, buff, buff_len);
 }
 
 // - Function sends data using sendto() and handles errors returned by it
